.github/workflows/build-and-test.yml

```diff
@@ -125,3 +132,207 @@ jobs:
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+      - name: Restore module cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-go-modules
+        with:
+          path: /home/runner/go/pkg/mod
+          key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('./go.mod') }}
+      - name: split loadtest jobs
+        id: splitloadtest
+        run: |
+          TESTS="$(make -s testbed-list-loadtest | xargs echo|sed 's/ /|/g')"
+          TESTS=(${TESTS//|/ })
+          MATRIX="{\"include\":["
+          curr=""
+          for i in "${!TESTS[@]}"; do
+            if (( i > 0 && i % 2 == 0 )); then
+              curr+="|${TESTS[$i]}"
+            else
+              if [ -n "$curr" ] && (( i>1 )); then
+                MATRIX+=",{\"test\":\"$curr\"}"
+              elif [ -n "$curr" ]; then
+                MATRIX+="{\"test\":\"$curr\"}"
+              fi
+              curr="${TESTS[$i]}"
+            fi
+          done
+          MATRIX+="]}"
+          echo "::set-output name=matrix::$MATRIX"
+  loadtest:
+    runs-on: ubuntu-latest
+    needs: [setup-environment, loadtestpre]
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.loadtestpre.outputs.matrix) }}
+    env:
+      TEST_RESULTS: testbed/tests/results/junit/results.xml
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.15
+      - name: Download tool binaries
+        uses: actions/download-artifact@v2
+        with:
+          name: tool-binaries
+          path: /home/runner/go/bin
+      - name: Add execute permissions to tool binaries
+        run: chmod -R +x /home/runner/go/bin
+      - name: Setup env
+        run: |
+          echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+      - name: Restore module cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-go-modules
+        with:
+          path: /home/runner/go/pkg/mod
+          key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('./go.mod') }}
+      - name: Install fluentbit
+        if: ${{ contains(matrix.test, 'Log10kDPS') }}
+        run: |
+          wget https://packages.fluentbit.io/ubuntu/bionic/pool/main/t/td-agent-bit/td-agent-bit_1.5.3_amd64.deb
+          sudo dpkg -i ./td-agent-bit*.deb
+          echo "/opt/td-agent-bit/bin" >> $GITHUB_PATH
+          sudo ln -s /opt/td-agent-bit/bin/td-agent-bit /usr/local/bin/fluent-bit
+      - run: mkdir -p results && touch results/TESTRESULTS.md
+      - name: Loadtest
+        run: make testbed-loadtest
+        env:
+          TEST_ARGS: "-test.run=${{ matrix.test }}"
+      - name: Create test result archive # some test results have invalid characters
+        if: ${{ failure() || success() }}
+        continue-on-error: true
+        run: tar -cvf test_results.tar testbed/tests/results
```

Comment:
- making a tarball of the test results because some tests have filenames with invalid characters (namely the `0*0bytes` test) which causes the upload action to fail. 

- `if: ${{ failure() || success() }}` this ensures that test results are uploaded even if the loadtest fails which can be useful for debugging

- `continue-on-error: true` Sometimes (rarely, but still) the upload action fails due to a network error on github's side. This makes sure the entire job doesnt fail if the test results fail to upload